server/game/cards/09.5-aCF/ArmyOfTheRisingWave.js

```diff
@@ -0,0 +1,20 @@
+const DrawCard = require('../../drawcard.js');
+const AbilityDsl = require('../../abilitydsl.js');
+
+class ArmyOfTheRisingWave extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            title: 'Place one fate on each unclaimed ring',
+            when: {
+                onCharacterEntersPlay: (event, context) => event.card === context.source
+            },
+            gameAction: AbilityDsl.actions.placeFateOnRing(() => ({
+                target: Object.values(this.game.rings).filter(ring => ring.isUnclaimed())
+            }))
```

Comment:
You probably looked at written in the stars and followed that, but PlaceFateOnRing also excepts a function that gets passed a context object. So you could do
`
gameAction: AbilityDsl.actions.placeFateOnRing(context => ({
       target: Object.values(context.game.rings).filter(ring => ring.isUnclaimed())
}))
`

Which just is a bit safer.